core/src/finding/details/debug_output.cpp

```diff
@@ -0,0 +1,36 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "traccc/finding/details/debug_output.hpp"
+
+#include "traccc/utils/logging.hpp"
+
+namespace traccc {
+void propagate_to_next_surface_debug::reset() {
+    for (std::size_t i = 0;
+         i < static_cast<std::size_t>(propagate_to_next_surface_exit_mode::MAX);
```

Comment:
Wouldn't `std::size(m_failure_count)` be more robust here?

Reply:
It's mainly to reduce the number of characters needed to express the same. I agree that it's not more robust code. But it's certainly shorter.

Reply:
Okay sure, I can use `std::size`. :+1: